memory/migrate_pages.py

```diff
@@ -88,9 +93,14 @@ def test(self):
         cmd = './node_move_pages -n %s' % self.nr_chunks
 
         if self.hpage:
-            cmd += ' -h'
-            if self.hpage_commit:
-                cmd += ' -o'
+           if (self.dist.name == 'rhel' and self.dist.version == '9'):
+               if self.hpage_commit:
+                  self.cancel("Hugepage(over commit) tests are cancelled on RHEL-9")
+               self.cancel("Hugepage tests are cancelled on RHEL-9")
+           else:
+              cmd += ' -h'
+              if self.hpage_commit:
+                 cmd += ' -o'
```

Comment:
These changes are not required. You can assign value of 0 to hpage_commit and hpage in setUp().

Reply:
IMO, these changes are required when we use yaml input.. where we can skip/cancel huge_page related tests.
Ex: run with yaml file

[root@xxxx memory]# avocado run --test-runner runner migrate_pages.py -m /home/xxxx/avocado-misc-tests/memory/migrate_pages.py.data/migrate_pages.yaml
JOB ID     : 939abd06bfebcee076cbfeaf99c647484bc5e439
JOB LOG    : /home/xxxx/tests/results/job-2022-10-20T12.23-939abd0/job.log
 (1/4) migrate_pages.py:MigratePages.test;run-page_type-normal_page-97e4: CANCEL: Environment prevents test! Check logs for node data (2.21 s)
 (2/4) migrate_pages.py:MigratePages.test;run-page_type-hugepage-d0ee: CANCEL: Hugepage tests are cancelled on RHEL-9 (1.01 s)
 (3/4) migrate_pages.py:MigratePages.test;run-page_type-overcommit_huge-c71d: CANCEL: Hugepage(over commit) tests are cancelled on RHEL-9 (1.03 s)
 (4/4) migrate_pages.py:MigratePages.test;run-page_type-thp-6ce9: CANCEL: Environment prevents test! Check logs for node data (3.91 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB HTML   : /home/xxxx/tests/results/job-2022-10-20T12.23-939abd0/results.html
JOB TIME   : 8.95 s

So with the YAML file, we will cancel huge pages related tests and run only 1,4 as @Geet2  initially suggested.
Currently, 1, and 4 are failed or canceled due to other functional issues(the same issue seen with Ubu too)... 
We will take up to debug this later(the 1, 4 test case issue)... while this PR gets reviewed and get merged.

Reply:
With the above changes, we don't have to handle yaml params in setup(keep the code intact), and instead with 2 lines of code to cancel the tests if -h or -o would be easier in the test function is what I feel.

Reply:
As per the changes suggested the c program when compiled on RHEL 9.0 will only enable THP and normal pages. -h and -o case code won't even be compiled and hence no need to cancel the tests as those aren't applicable.

Anyway I will leave it to the approvers to take a final call.